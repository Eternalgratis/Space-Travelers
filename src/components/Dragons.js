import { React, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import DragonItem from './DragonItem';
import { getDragons } from '../redux/dragons/dragons';

const Dragons = () => {
  const dispatch = useDispatch();
  const dragons = useSelector((state) => state.dragons);

  useEffect(() => {
    if (!dragons.length) {
      dispatch(getDragons());
    }
  }, []);

  return (
    <div className="dragonContainer">
      {dragons.map((dragon) => (
        <DragonItem dragon={dragon} key={dragon.id} />
      ))}
    </div>
  );
};

export default Dragons;
