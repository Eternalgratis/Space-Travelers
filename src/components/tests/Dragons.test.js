import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import React from 'react';
import store from '../../redux/configureStore';
import Dragons from '../Dragons';

describe('Dragons component', () => {
  it('Renders Dragons component', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Dragons />
        </Provider>,
      );
    expect(tree).toMatchSnapshot();
  });
});
