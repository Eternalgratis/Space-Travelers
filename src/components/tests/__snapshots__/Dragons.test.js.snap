// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dragons component Renders Dragons component 1`] = `
<div
  className="dragonContainer"
/>
`;
